src/runtime/LoadingView.tsx

```diff
@@ -17,31 +18,43 @@ export default class LoadingView extends React.Component<Props> {
       const error = value["_error"].message;
 
       // make sure the error isn't a duplicate
-      const found = this.errors.findIndex((err) => 
-        err === error
-      ) > -1;
+      const found = this.errors.indexOf(error) > -1;
 
       if (!found) {
         this.errors.push(error);
       }
     }
-  }
+  };
 
   public render() {
     const { logs } = this.props;
 
     R.forEachObjIndexed(this.findErrorKeys, logs);
     return (
-      <Popup
-        trigger={<Spinner name='double-bounce'/>}
-        position="right center"
-        on="hover"
-        contentStyle={{ width: "auto", display: "flex", flexWrap: "wrap" }}
-      >
-        <div>
-          {this.errors.map(error => error)}
-        </div>
-      </Popup>
-    )
+      <Loader.Render>
+        {(
+          CustomLoader:
+            | React.ComponentClass<any>
+            | React.StatelessComponent<any>
+        ) =>
+          CustomLoader ? (
+            <CustomLoader errors={this.errors} />
```

Comment:
I thought that rendering this way and making the user pass the component as `render={MyComponent}` makes the code cleaner - Lmk your thoughts

Reply:
While it is simpler, it doesn't explicity let the user know what props are available for their component. If they instead are expected to provide a function with a typed props arg, they can then use the errors in the props & render them how they see fit. In the above example they couldn't let's say pass custom props to MyComponent.

I wasn't sure what best practice was for this, so I looked to how other projects do this (formik, react-router, etc): https://reactjs.org/docs/render-props.html